src/docker.rs

```diff
@@ -183,15 +185,36 @@ pub fn run(target: &Target,
     }
 
     docker
-        .arg(&image(toml, target)?)
+        .arg(&image(toml, target, verbose)?)
         .args(&["sh", "-c", &format!("PATH=$PATH:/rust/bin {:?}", cmd)])
         .run_and_get_status(verbose)
 }
 
-pub fn image(toml: Option<&Toml>, target: &Target) -> Result<String> {
+fn build_docker_image(toml: &Toml, target: &Target, verbose: bool) -> Result<String> {
+    let mut docker = docker_command("build")?;
+    let context = toml.context(target)?.map(|s| s.to_owned()).unwrap();
+    docker.arg(context);
+    if let Some(dockerfile) = toml.dockerfile(target)? {
+        docker.args(&["-f", dockerfile]);
+    }
+    if verbose {
+        // Running `docker build` twice in quick succession is cheap
+        // (~0.1s extra) because of docker's layer cache.
+        // Here, we run it once without --quiet, to get the verbose output,
+        // then run it again with --quiet to get the image sha.
+        docker.run(verbose)?
+    }
+    let mut image_id = docker.arg("--quiet").run_and_get_stdout(verbose)?;
```

Comment:
You should add the `--iidfile` option to the first `docker build` to write the image id to a temp file, instead of running it again and trying to parse the human-readable output.

Reply:
That seems to work. Thanks. Pushed.